ShaderView improvements

johnhaddon opened the pull request:
This improves UI responsiveness when viewing a shader ball for the first time. When this happens, there can be a short delay while the renderer starts up, but (at least for Arnold) this occurs on the background rendering thread. Unfortunately we were performing redundant scene updates immediately after requesting startup, and to perform an update we need to pause the renderer, which means requesting an abort from the UI thread and then waiting to join the background rendering thread. This locked the main thread until the renderer had started up.

Eliminating the redundant updates doesn't make the renderer start up any faster, but does at least mean that the main UI thread is responsive in the meantime.

andrewkaufman commented:
I'm finding this much better in general, but still quite slow for alSurface. I have a feeling its largely todo with the massive number of widgets that we need to make, combined with the fact that the IPR is using up all my cores. I found that setting ai:threads to -3 felt pretty much back to where we stared (at least on my box).

```
diff --git a/python/GafferArnold/ArnoldShaderBall.py b/python/GafferArnold/ArnoldShaderBall.py
index 74ac84f..f59fe50 100644
--- a/python/GafferArnold/ArnoldShaderBall.py
+++ b/python/GafferArnold/ArnoldShaderBall.py
@@ -66,6 +66,8 @@ class ArnoldShaderBall( GafferScene.ShaderBall ) :
                self["__arnoldOptions"]["in"].setInput( self["__parentLights"]["out"] )
                self["__arnoldOptions"]["options"]["aaSamples"]["enabled"].setValue( True )
                self["__arnoldOptions"]["options"]["aaSamples"]["value"].setValue( 3 )
+               self["__arnoldOptions"]["options"]["threads"]["enabled"].setValue( True )
+               self["__arnoldOptions"]["options"]["threads"]["value"].setValue( -3 )
 ```

johnhaddon commented:
I've updated this with the thread limiting you suggested (seems like a good improvement at my end too), and added the ability to change scene registrations on the fly.
